worker/host.go

```diff
@@ -217,44 +217,56 @@ func (h *host) FetchPriceTable(ctx context.Context, rev *types.FileContractRevis
 }
 
 func (h *host) FundAccount(ctx context.Context, balance types.Currency, rev *types.FileContractRevision) error {
-	// fetch pricetable
-	pt, err := h.priceTable(ctx, rev)
-	if err != nil {
-		return err
-	}
-
-	// calculate the amount to deposit
+	// fetch current balance
 	curr, err := h.acc.Balance(ctx)
 	if err != nil {
 		return err
 	}
+
+	// return early if we have the desired balance
 	if curr.Cmp(balance) >= 0 {
 		return nil
 	}
-	amount := balance.Sub(curr)
-
-	// cap the amount by the amount of money left in the contract
-	renterFunds := rev.ValidRenterPayout()
-	possibleFundCost := pt.FundAccountCost.Add(pt.UpdatePriceTableCost)
-	if renterFunds.Cmp(possibleFundCost) <= 0 {
-		return fmt.Errorf("insufficient funds to fund account: %v <= %v", renterFunds, possibleFundCost)
-	} else if maxAmount := renterFunds.Sub(possibleFundCost); maxAmount.Cmp(amount) < 0 {
-		amount = maxAmount
-	}
 
 	return h.acc.WithDeposit(ctx, func() (types.Currency, error) {
-		return amount, h.transportPool.withTransportV3(ctx, h.hk, h.siamuxAddr, func(ctx context.Context, t *transportV3) (err error) {
-			cost := amount.Add(pt.FundAccountCost)
-			payment, err := payByContract(rev, cost, rhpv3.Account{}, h.renterKey) // no account needed for funding
+		deposit := balance.Sub(curr)
```

Comment:
This is nicer directly under the `if curr.Cmp(balance) >= 0 {` check imo.

Reply:
yeah went back and forth, will move